lib/cli/src/commands/app/logs.rs

```diff
@@ -113,7 +113,7 @@ impl crate::commands::AsyncCliCommand for CmdAppLogs {
         while let Some(logs) = logs_stream.next().await {
             let logs = logs?;
 
-            let limit = std::cmp::max(logs.len(), rem);
+            let limit = std::cmp::min(logs.len(), rem);
 
             let logs = &logs[..limit];
```

Comment:
Let's prevent a similar issue in the future by not using slice indexing.

Actually, why are we indexing here at all?
The limit is just supposed to be for querying, why would we discard logs returned by the API?

Reply:
It's not just for querying. Since there is no direct relation between `max` and the maximum amount of logs returned by the backend, we have to discard logs somewhere in order to respect `max`. For example, if user wants to see at most 30 logs, and backend returns 20 at a time, we have to discard 10 logs in the second batch.

IMO Indexing is fine as long as its guard is correct. Plus, using `get` here would swallow this logical bug. This is a type of bug that should've been caught in a test, or during review. But I'm fine with using `get` as well.

Reply:
Indexing can panic, and it has panicked here and can again if we change the code somehow. 

There's a reason why iterators exist, let's just use those.
It can also just be `slice.get(0..limit)`, which doesn't panic.